### DISCUSSION TITLE
Using pytest-bdd with pytest-asyncio

### DISCUSSION THREAD

**Discussion by MAMuftuoglu** [pytest-dev/pytest-bdd#622] [Q&A] [answered]
I am trying to create a test for set some values in a device and then query them from device to assert that they are equal several times with specific time intervals. But for this to work I need to run query and assert things async because I cannot wait for a response to continue. But I couldn't use `then` functions with `pytest.mark.asyncio`. How can I use both of them at the same time?

Funtion is like this (I couldn't make a multi line code snippet)

@pytest.mark.asyncio
@then(parsers.parse("The query result for {setting} is {value:n}"))
@then(parsers.parse("The query result for {setting} is {value:bool}", extra_types={"bool": bool}))
async def test_get_value(destination, setting, value):
    responses = await run_queries(destination, setting, CONNECTION_FILE)
    for response in responses:
        assert response == value

**Comment by MAMuftuoglu** [comment]
I realized that when I add any decorator related to bdd, it just doesn't recognize other pytest decorators.

**Comment by youtux** [comment, marked as answer]
check this out: https://github.com/pytest-dev/pytest-bdd/issues/223#issuecomment-1647998987
